cloudinit/sources/helpers/azure.py

```diff
@@ -330,34 +330,22 @@ def get_ip_from_lease_value(fallback_lease_value):
 
 
 @azure_ds_telemetry_reporter
-def http_with_retries(url, **kwargs) -> url_helper.UrlResponse:
+def http_with_retries(
+    url, *, headers: dict, data=None
```

Comment:
Since we are type hinting, let's get url too
```suggestion
    url: str, *, headers: dict, data=None
```

Reply:
done, for data too, thanks!